### PR TITLE
Use SaltStack repos

### PR DESCRIPTION
**Pull request by stbenjam**
Starting with 2015.8.0, SaltStack will provide their own repositories.

I'll update for Debian later, as Ubuntu is not yet available.  https://groups.google.com/forum/#!topic/salt-users/hteT6sDi39w

### CODE REVIEW

File: kickstart/provision.erb [hunk 1]
```diff
@@ -79,6 +80,9 @@ repo --name="EPEL" --mirrorlist=https://mirrors.fedoraproject.org/metalink?repo=
 repo --name=puppetlabs-products --baseurl=http://yum.puppetlabs.com/el/<%= @host.operatingsystem.major %>/products/<%= @host.architecture %><%= proxy_string %>
 repo --name=puppetlabs-deps --baseurl=http://yum.puppetlabs.com/el/<%= @host.operatingsystem.major %>/dependencies/<%= @host.architecture %><%= proxy_string %>
 <% end -%>
+<% if salt_enabled && @host.params['enable-saltstack-repo'] && @host.params['enable-saltstack-repo'] == 'true' -%>
+repo --name=saltstack-repo --baseurl=https://repo.saltstack.com/yum/rhel$releasever<%= @host.operatingsystem.major %><%= proxy_string %>
```
Comment (domcleal) [inline kickstart/provision.erb]: Did you mean to use both releasever and host.os.major?

**Comment by mmoll** [comment]
As it seems, only Debian/jessie is available on https://repo.saltstack.com/ while with `enable-saltstack-repo = true` http://debian.saltstack.com/ (holding 2015.5.3) is used, which also has no Ubuntu packages, so if this is getting relevant, we need to do some research/asking around anyway.

File: kickstart/provision.erb (hunk 1 above)
Comment (mmoll) [inline kickstart/provision.erb]: @stbenjam ping?

File: kickstart/finish.erb [hunk 2]
```diff
@@ -44,6 +45,27 @@ rpm -ivh http://yum.puppetlabs.com/puppetlabs-release-el-<%= os_major -%>.noarch
...
 
+<% if salt_enabled %>
+<% if @host.params['enable-saltstack-repo'] && @host.params['enable-saltstack-repo'] == 'true' -%>
+cat > /etc/yum.repos.d/saltstack.repo <<EOF
+[saltstack-repo]
+name=SaltStack repo for RHEL/CentOS $releasever
+baseurl=https://repo.saltstack.com/yum/rhel$releasever
+enabled=1
+gpgcheck=1
+gpgkey=https://repo.saltstack.com/yum/rhel$releasever/SALTSTACK-GPG-KEY.pub
```
Comment (lzap) [inline kickstart/finish.erb:56]: Will this work on CentOS 6 or even 5? Separate wget could be safer.

**Comment by lzap** [comment]
Looks good here, except during provisioning of current stable Salt I get into infinite loop of errors in the post scriplet, hitting this: https://github.com/saltstack/salt/issues/27453

```
mkdir /etc/salt/pki
```

Should fix this but it does not.

**Comment by stbenjam** [comment]
@lzap What about /etc/salt/pki/minion?

**Comment by lzap** [comment]
That seemed to work!

**Comment by lzap** [comment]
Wait, I get this now with CentOS 7.1

screenshot_el7 local lan_2015-10-09_22 41 27

**Comment by stbenjam** [comment]
@lzap What was the resolution to the problem you saw? I tested the snippet it and salt-minion gets installed fine.

**Comment by stbenjam** [comment]
Also, `mkdir -p /etc/salt/pki/minion` in the snippet does work around saltstack/salt#27453 for me.  Don't think I want to include that unless they won't fix the issue in a way that continues to work with calling salt-call before the service is started.

**Comment by lzap** [comment]
Yes, fixed. The other prooblem was missing EPEL.

**Comment by mmoll** [comment]
So, we're waiting for that saltstack issue to be fixed in these repos?

**Comment by stbenjam** [comment]
Yup, I don't want to merge it until its fixed, EPEL is older version but should still work fine

**Comment by mmoll** [comment]
OK, thanks. :hourglass_flowing_sand:

**Comment by mmoll** [comment]
saltstack/salt#29701 (issue) and saltstack/salt#30371 (pr) are the current links for that problem

**Comment by mmoll** [comment]
the upstream problem may be fixed with 2015.8.8, so this should be rebased and tested again :)

**Comment by mmoll** [comment]
@stbenjam I'd suggest to close this for now and reimplement in a snippet if needed like #362

**Comment by mmoll** [comment]
closing due to inactivity

**Pull request closed without merging**